service/api.go

```diff
@@ -381,6 +382,47 @@ func (as *APIServer) AttachFiles(w http.ResponseWriter, r *http.Request) {
 	gimlet.WriteJSON(w, fmt.Sprintf("Artifact files for task %v successfully attached", t.Id))
 }
 
+// SetDownstreamParams updates file mappings for a task or build
+func (as *APIServer) SetDownstreamParams(w http.ResponseWriter, r *http.Request) {
+	t := MustHaveTask(r)
+	grip.Infoln("Setting downstream expansions for task:", t.Id)
+
+	var downstreamParams []patch.Parameter
+	err := utility.ReadJSON(util.NewRequestReader(r), &downstreamParams)
+	if err != nil {
+		errorMessage := fmt.Sprintf("Error reading downstream expansions for task %v: %v", t.Id, err)
+		grip.Error(message.Fields{
+			"message": errorMessage,
+			"task_id": t.Id,
+		})
+		gimlet.WriteJSONError(w, errorMessage)
+		return
+	}
+	p, err := patch.FindOne(patch.ByVersion(t.Version))
+
+	if err != nil {
```

Comment:
I think it would probably be best to also check if `p == nil` (although it seams unlikely).

Reply:
just following up on this one